### PR TITLE
Add DGEB evaluation integration for UME models

### PR DESCRIPTION
**Pull request by ncfrey**
## Summary

This PR adds comprehensive DGEB (DNA/Protein Language Model Benchmark) evaluation integration for UME models, enabling standardized benchmarking of biological sequence models.

### Key Features

- **DGEB Integration**: Full integration with the DGEB framework for protein and DNA sequence benchmarking
- **UME Model Support**: Direct support for UME models via `Ume.from_pretrained()` method
- **Local Evaluation**: Run evaluations locally without requiring Hugging Face Hub integration
- **Comprehensive Reporting**: Generate both JSON and Markdown reports for easy sharing
- **CLI Interface**: New `lobster_dgeb_eval` command for easy evaluation runs
- **Complete Test Suite**: Full test coverage with fixtures for both pretrained and checkpoint models

### Implementation Details

- **UMEAdapter Class**: Bridges UME models with DGEB's `BioSeqTransformer` interface
- **Evaluation Runner**: Orchestrates evaluation across multiple tasks with proper logging
- **Report Generation**: Creates detailed reports with task summaries and model metadata
- **Error Handling**: Robust error handling for model loading and evaluation failures
- **Device Management**: Automatic CPU/GPU device detection and management

### Files Added

- `src/lobster/evaluation/dgeb_adapter.py` - Core adapter implementation
- `src/lobster/evaluation/dgeb_runner.py` - Evaluation orchestration
- `src/lobster/evaluation/README.md` - Comprehensive documentation
- `src/lobster/cmdline/dgeb_eval.py` - CLI entry point
- `tests/lobster/evaluation/test_dgeb_integration.py` - Complete test suite

### Usage

```bash
# Evaluate a pretrained model on protein tasks
lobster_dgeb_eval ume-mini-base-12M --modality protein

# Evaluate on specific tasks
lobster_dgeb_eval ume-mini-base-12M --modality protein --tasks ProteinKNN ProteinGym

# Evaluate from checkpoint
lobster_dgeb_eval /path/to/checkpoint.ckpt --modality dna --output-dir my_results
```

### Testing

All tests pass including:
- DGEB task loading and compatibility
- UME model adapter functionality
- Embedding generation and shape validation
- Error handling for edge cases
- Both pretrained and checkpoint model support

## Test Plan

- [x] Unit tests for adapter functionality
- [x] Integration tests with DGEB framework
- [x] CLI interface testing
- [x] Report generation validation
- [x] Error handling verification
- [x] Documentation completeness check

### CODE REVIEW

File: src/lobster/evaluation/dgeb_adapter.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            try:
+                # Get embeddings from UME model
+                batch_embeddings = self.model.embed_sequences(
+                    batch_sequences,
+                    modality=lobster_modality,
+                    aggregate=True,  # Use mean pooling
+                )
+
+                # Convert to numpy
+                batch_embeddings = batch_embeddings.detach().cpu().numpy()
```
Comment (ncfrey) [inline src/lobster/evaluation/dgeb_adapter.py:192]: TODO: run on gpu as well

File: src/lobster/evaluation/README.md [hunk 2]
```diff
@@ -115,6 +115,178 @@ Evaluation date: 2023-08-15 14:30:22
...
+### Configuration Options
+
+```bash
+uv run lobster_dgeb_eval MODEL_NAME \
+    --modality {protein,dna}           # Required: sequence modality
+    --tasks TASK1 TASK2 ...            # Optional: specific tasks (default: all)
+    --output-dir OUTPUT_DIR            # Optional: results directory (default: dgeb_results)
+    --batch-size BATCH_SIZE            # Optional: encoding batch size (default: 32)
+    --max-seq-length MAX_LENGTH        # Optional: max sequence length (default: 1024)
+    --use-flash-attn                   # Optional: enable flash attention
```
Comment (taylormjs) [inline src/lobster/evaluation/README.md:242]: We may only want one of these two flags `--use-flash-attn` or `--no-flash-attn`

File: src/lobster/evaluation/README.md [hunk 3]
```diff
@@ -115,6 +115,178 @@ Evaluation date: 2023-08-15 14:30:22
...
+    print(f'Tasks: {len(results[\"results\"])}')
+    for task in results['results']:
+        print(f'  {task[\"task_name\"]}: {task[\"scores\"]}')
+"
+```
+
+### Performance Tips
+
+- **Batch Size**: Increase `--batch-size` for faster evaluation on GPU (try 64-128)
+- **Sequence Length**: Reduce `--max-seq-length` if memory is limited (try 512)
```
Comment (taylormjs) [inline src/lobster/evaluation/README.md:278]: We should consider indicating if any tasks require longer sequence lengths, especially for DNA tasks. Having a flashback to the AAV task where all sequence variation was after index 512

File: src/lobster/evaluation/dgeb_adapter.py [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        modality : LobsterModality
+            The biological modality for tokenization.
+
+        Returns
+        -------
+        np.ndarray
+            Embeddings of shape [batch_size, num_layers, embedding_dim].
+        """
+        # For now, use the high-level embed_sequences method which gives us aggregated embeddings
+        # TODO: In the future, we could implement proper layer-wise extraction by calling
```
Comment (taylormjs) [inline src/lobster/evaluation/dgeb_adapter.py:337]: Agreed! This is good for now

File: src/lobster/evaluation/dgeb_adapter.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+from dgeb.models import BioSeqTransformer
+from dgeb.modality import Modality
+
+from lobster.constants import Modality as LobsterModality
+from lobster.model import UME
+
+logger = logging.getLogger(__name__)
+
+
+class UMEAdapter(BioSeqTransformer):
```
Comment (taylormjs) [inline src/lobster/evaluation/dgeb_adapter.py:17]: Maybe change to UMEAdapterDGEB or something more specific? In case we add more of these, and to avoid confusion in case we add LoRAdapters

File: src/lobster/evaluation/dgeb_runner.py [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+    """Generate a human-readable evaluation report.
+
+    Parameters
+    ----------
+    results_summary : dict[str, Any]
+        Results summary from run_evaluation.
+    output_path : Path
+        Path to save the report.
+    """
+    report_path = output_path / "evaluation_report.md"
```
Comment (taylormjs) [inline src/lobster/evaluation/dgeb_runner.py:183]: Should this include a timestamp to avoid overwriting reports? I suppose that could also be in the output_path

File: tests/lobster/evaluation/test_dgeb_integration.py [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+        temp_dir.rmdir()
+
+
+def test_dgeb_tasks():
+    """Test that DGEB tasks can be loaded."""
+    import dgeb
+
+    # Get available tasks
+    all_tasks = dgeb.get_all_task_names()
+    assert len(all_tasks) > 0, "Should find some tasks"
```
Comment (taylormjs) [inline tests/lobster/evaluation/test_dgeb_integration.py:82]: Assert `len(all_tasks) = num_tasks` instead of > 0? Just to be sure we're capturing all of them. Same for protein & dna tasks

File: tests/lobster/model/modern_bert/test_rotary.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    # The rotary embedding should modify the first 'dim' dimensions of each head
+    diff = torch.abs(output - qkv_original)
+    max_diff = torch.max(diff)
+    print(f"Maximum difference between output and input: {max_diff}")
+
+    # Check that at least some values changed (rotary embedding was applied)
+    assert max_diff > 1e-6, f"Rotary embedding did not modify the tensor. Max diff: {max_diff}"
+
+
+def test_rotary_embedding_positional_invariance():
```
Comment (taylormjs) [inline tests/lobster/model/modern_bert/test_rotary.py:83]: Nice test case!

**Review by taylormjs — APPROVED** [review]
Looks great overall! Left some minor comments

**Comment by taylormjs** [comment]
@ncfrey Just saw the merge conflicts in uv.lock

File: tests/lobster/evaluation/test_dgeb_integration.py (hunk 7 above)
Comment (ncfrey) [reply tests/lobster/evaluation/test_dgeb_integration.py:82]: kept this as > 0 to be robust to dgeb dataset updates

File: src/lobster/cmdline/dgeb_eval.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+"""Command-line interface for DGEB evaluation of UME models."""
+
+import sys
+from pathlib import Path
+
+# Add the evaluation module to the path
+sys.path.insert(0, str(Path(__file__).parent.parent))
```
Comment (karinazad) [inline src/lobster/cmdline/dgeb_eval.py:7]: we probably want to get rid of this? @ncfrey
